src/Rbac/DefaultRoleManager/RoleManager.php

```diff
@@ -55,6 +87,17 @@ protected function hasRole(string $name): bool
      */
     protected function createRole(string $name): Role
     {
+        if ($this->hasPattern) {
+            foreach ($this->allRoles as $key => $value) {
+                $func = $this->matchingFunc;
+                if ($func($name, $key)) {
+                    $name = $key;
+
```

Comment:
Refer to: https://github.com/casbin/casbin/blob/master/rbac/default-role-manager/role_manager.go#L274